src/Internal/SetupDecryptionHandler.cs

```diff
@@ -429,14 +429,20 @@ protected void DecryptObjectUsingMetadata(GetObjectResponse getObjectResponse, b
                         && EncryptionUtils.XAmzAesGcmCekAlgValue.Equals(instructions.MaterialsDescription[EncryptionUtils.XAmzEncryptionContextCekAlg]))
                     {
                         // Decrypt the object with V2 instruction
-                        EncryptionUtils.DecryptObjectUsingInstructionsV2(getObjectResponse, instructions);
+                        EncryptionUtils.DecryptObjectUsingInstructionsGcm(getObjectResponse, instructions);
                     }
                     else
                     {
                         throw new AmazonCryptoException($"The content encryption algorithm used at encryption time does not match the algorithm stored for decryption time." +
                                                         " The object may be altered or corrupted.");
                     }
                 }
+                // this else if was added to address AWS SES decryption use case: https://github.com/aws/amazon-s3-encryption-client-dotnet/issues/26
```

Comment:
nit: this could be more descriptive, I would prefer something like: 

```
// Handle legacy KMS (v1) mode with GCM content encryption 
// See $LINK for context 
```

Since Github won't let me put comments on lines which have no diff, I will offer a few more comment suggestions to make the surrounding code easier to understand. 

Above the if statement on L425,L426: `Check if encryption context is present for kms+context (v2) objects`
L428,429: `If encryption context is present, ensure that the GCM algorithm name is in the EC as expected in v2`

Optional - above L423: `Mode is either kms (v1) or kms+context (v2)`

Reply:
Thank you! Will add the comments